Fixed the names of Symbol Attribute fields

MirceaS opened the pull request:
Fixes #2222 

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

ttuegel commented on kore/src/Kore/Attribute/Symbol.hs:
```diff
@@ -108,29 +108,29 @@ view the effective attributes, use the functions defined in this module.
+    , klabel         :: !Klabel
+    , symbolKywd     :: !SymbolKywd
+    , noEvaluators   :: !NoEvaluators
+    , builtinUnit    :: !(Unit SymbolOrAlias)
+    , builtinElement :: !(Element SymbolOrAlias)
+    , builtinConcat  :: !(Concat SymbolOrAlias)
+    -- ^ The above three are only populated if the symbol is a builtinAc symbol
```
I don't understand what is "built-in" about these symbols. They are user-defined symbols only used for unparsing. (Now that I think of it, the comment should mention these are only used for unparsing.) If simply `unit`, `element`, and `concat` cause too many collisions, the next best names would be `unitSymbol`, etc. As for the documentation, what is a "builtinAc symbol"?

MirceaS replied:
The fields are "built-in" in the sense that they represent the built-in symbol labels but I don't mind changing them if we find a better name for them. `unitSymbol`  and the rest are also unavailable due to import conflicts from `Unit`, `Element` and `Concat`.
And a built-in Ac symbol is a symbol of a built-in associative-commutative type. I tried to use the same naming convention that we use for `InternalAc`, NormalizedAc` etc.

MirceaS commented:
@ttuegel I addressed your comments with my latest commit but I wanna use this branch to also fix some errors that Virgil noticed related to the original PR, so I'll keep it open a little longer.

MirceaS commented:
#2222
